### PR TITLE
OT99-41-Endpoint-Category-Details

### PR DESCRIPTION
**Pull request by ingrid-inga**
================================Get by id.================================================

Screen Shot 2021-11-10 at 13 16 32

================================404 Not Found.=========================================

Screen Shot 2021-11-10 at 13 17 18

### CODE REVIEW

File: src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java [hunk 1]
```diff
@@ -29,4 +35,29 @@
...
+        categoryDto.setUpdatedAt(LocalDateTime.now());
+
+        return categoryRepository.save(category);
+    }
+
+
+	@Override
+	public Category getCategoryById(long id) throws NotFoundException {
+		if(categoryRepository.existsById(id)){
+			Category category = categoryMapper.dtoToEntity(id);
```
Comment (nicolascareri) [inline src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java:55]: Estas mapeando una category y despues no la usas

File: src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java [hunk 2]
```diff
@@ -29,4 +35,29 @@
...
+
+		return categoryRepository.findById(id).get();
+		}else{
+		   throw new NotFoundException("Wrong!, Category doesn't exist.");
```
Comment (nicolascareri) [inline src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java:59]: yo en este caso usaria el orElseThrow que te provee el optional y te ahorras los IF y el ELSE

**Review by nicolascareri — CHANGES_REQUESTED** [review]
Solucionar conflictos

File: src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java (hunk 1 above)
Comment (nicolascareri) [reply src/main/java/com/alkemy/ong/service/impl/CategoryServiceImpl.java:55]: public Category getCategoryById(long id) throws NotFoundException {
  Category category = categoryRepository.findById(id).orElseThrow( ()-> NotFoundException("Wrong!, Category doesn't exist.")  );
return categoryMapper.entityToDto(category);
}

File: src/main/java/com/alkemy/ong/mapper/CategoryMapper.java [hunk 3]
```diff
@@ -23,5 +30,21 @@
...
+		Category category = modelMapper.map(catDto, Category.class);
+		  return category;
+		}
+
+	public Category toCategoryDetailsDto(long id) {
+			Category category = modelMapper.map(id, Category.class);
+			return category;
+		}
+
+	public Category dtoToEntity(long id) {
```
Comment (nicolascareri) [inline src/main/java/com/alkemy/ong/mapper/CategoryMapper.java:45]: public CategoryDto EntityToDto (Category category) {
 CategoryDto categoryDto = modelMapper.map(category, CategoryDto.class);
 return categoryDto;
}

**Review by nicolascareri — CHANGES_REQUESTED** [review]

**Comment by ingrid-inga** [comment]
****Se actualiza "Time" del código correspondiente a "Create" (Referencia: PR - OT99-42. Link: https://github.com/alkemyTech/OT-99-Server/pull/28) . Anda ok!.

Screen Shot 2021-11-11 at 00 09 13

=========================================================================================

*******Retorna los datos de la categoría creada by Id.

Screen Shot 2021-11-11 at 00 10 03

**Review by ingrid-inga — COMMENTED** [review]
Double code in Error Handler. Updated.

**Review by nicolascareri — APPROVED** [review]

**Pull request merged**